src/Generation/EmulatorSupportGenerator.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+declare(strict_types=1);
+
+namespace Google\Generator\Generation;
+
+use Google\Generator\Ast\AST;
+use Google\Generator\Ast\Access;
+use Google\Generator\Ast\PhpDoc;
+use Google\Generator\Utils\Type;
+
+class EmulatorSupportGenerator
+{
+    /**
+     * Map of emulator support required clients and their expected env variables.
+     */
+    private static $emulatorSupportClients = [
+        'Google\Cloud\Spanner\Admin\Database\V1\Client\DatabaseAdminClient' => 'SPANNER_EMULATOR_HOST',
```

Comment:
What is the logic behind using a specific env var for different products? Why not just support a `GOOGLE_CLOUD_EMULATOR_HOST` which can work across products?

Reply:
This page specifically name it like `SPANNER_EMULATOR_HOST` : https://cloud.google.com/spanner/docs/emulator

Reply:
I know it's the officially supported one, I was just thinking that it's configured the same way for every product, so why have specific product names for it? But That's out of the scope of this PR.